Produce consume tutorial in scala (#459)
confluentinc/kafka-tutorials: _includes/tutorials/produce-consume-lang/scala/markup/dev/3_4-schema-serde.adoc

> The case class instances in our code will be serialized in an https://avro.apache.org/[Avro] format.

Reviewer: For your consideration: This is a great description of what's going on with the serdes. But there's a lot of information to absorb in the `Create a schema for the events` section. Maybe break the serde part into a separate section in the `scala.yml` file?

> The case class instances in our code will be serialized in an https://avro.apache.org/[Avro] format.

Reviewer 2: ```suggestion
In this tutorial, events will be will be serialized in https://avro.apache.org/[Avro] format.
```

> To do so, we will need the appropriate serializers/deserializers.

Reviewer 2: ```suggestion
To produce and consume events we need the appropriate serializers/deserializers.
```

> We add the following function to easily create a `Serializer[T]` where `T` in the type of the key or the value

Reviewer: Missing the path and name for creating the `ScalaReflectionSerde.scala` file in the text.

> We add the following function to easily create a `Serializer[T]` where `T` in the type of the key or the value

Reviewer 2: ```suggestion
We add the following function to easily create a `Serializer[T]` where `T` in the type of the key or the value we'd like to serialise. 
```

> we'd like to serialise. Add the class: `src/main/scala/io/confluent/developer/schema/ScalaReflectionSerde.scala`

Reviewer 2: ```suggestion
Add the following class: `src/main/scala/io/confluent/developer/schema/ScalaReflectionSerde.scala`
```

+++++
<pre class="snippet"><code class="scala">{%
    include_raw tutorials/produce-consume-lang/scala/code/src/main/scala/io/confluent/developer/schema/ScalaReflectionSerde.scala
%}</code></pre>
+++++

> While the specific method (where code is generated from schemas) for creating record is more common,
> in scala you might want to use the reflection method (where schemas is generated from code)
> to preserve your case class flexibility. Since Confluent 5.4.0, a
> `https://docs.confluent.io/current/streams/developer-guide/datatypes.html#reflection-avro[ReflectionAvroSerde]` class
> let you do that, but using it in scala might result in a incomplete type management. The previous file shows
> a quick way to cover this need in scala by using https://github.com/sksamuel/avro4s[Avro4s].
> Avro4s is a scala library generation of Apache Avro objects. It will create records or instances in two steps:
> 
> - Serialization:    Raw bytes <- Generic Records <- Case Class Instances
> - Deserialization:  Raw bytes -> Generic Records -> Case Class Instances

Reviewer 2: Since there are many opinons on techniques with this, I recommend we rework this slightly by just focusing on the method we are using.  The blurb about the ReflectionAvroSerde resulting in "a incomplete type management" is a little vague leaving the reader wondering about the issue is.

I suggest we remove that and simplify the section.  I'll make an attempt at that here but please feel free to rework as you see fit:

```suggestion
In this tutorial we are going to use a reflection based serialization method using https://github.com/sksamuel/avro4s[Avro4s], which is a serializing/deserializing library for Avro written in Scala.  

Avro4s performs it's de/serialization in multiple stages:
- For Serialization:      `Case Class Instance   ->  Generic Record -> Avro Formatted Bytes`
- For Deserialization:  `Avro Formatted bytes -> Generic Record -> Case Class Instance`
```

Author: Hi @rspurgeon,   
Honestly, I just tried to give everything I've got here (thinking: the more, the better)
But if this version says enough to you I think it's perfectly fine!!
